woocommerce/woocommerce-ios
# Add selected state UI and enable accessory view for product cell

Prerequisite for adding grouped products in #2199 

## Changes

- Added support for accessory view to the product cell by updating the bottom border's trailing constraint to pin to the cell instead of the cell content view
  - Otherwise, the bottom border ends before the leading edge of the accessory view like the screenshot below:
<img width="320" alt="Screen Shot 2020-06-29 at 9 02 52 PM" src="https://user-images.githubusercontent.com/1945542/86008826-eb3a4700-ba4b-11ea-9140-053822cf7b75.png">

- Added selected state to the product cell:
  - In its view model `ProductsTabProductViewModel `, added `isSelected: Bool` property that is default to `false`
  - Added an optional property `selectedProductImageOverlayView: UIView?`
  - Created an icon `UIImage.checkmarkInCellImageOverlay` with the size and color for display in `ProductsTabProductTableViewCell`
- Enabled SwiftUI preview for :
  - Created an initializer with default properties for `Product` in debug builds only
  - Added SwiftUI preview with non-selected and selected states to `ProductsTabProductViewModel`

## Testing

Currently `ProductsTabProductTableViewCell` is used in 3 places, and there should no visual changes to them:
1. Products tab
2. Product search results
3. Product variation list

- Go to the Products tab --> the loading state and list view should be the same as in `develop`
- Tap on the search icon --> the search results should look the same as in `develop`
- Tap on a variable product, then tap on the variations row --> the product variation list should be the same as in `develop`

## Example screenshots

SwiftUI preview for the non-selected and selected states in Light/Dark/Large font modes (thanks to @shiki's example earlier!). The selected state will be used in a future PR for #2199. Dynamic type isn't fully supported for the product cell yet.

<img width="395" alt="Screen Shot 2020-06-29 at 9 06 49 PM" src="https://user-images.githubusercontent.com/1945542/86009194-7ddae600-ba4c-11ea-86ef-0bab2bdd53c0.png">

Note: **no changes** are expected in the before and after screenshots.

\ | before | after
-- | -- | --
Products tab - light | ![Simulator Screen Shot - iPhone 11 - 2020-06-29 at 17 02 10](https://user-images.githubusercontent.com/1945542/86009314-a7940d00-ba4c-11ea-9010-1c8a6371dee1.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-29 at 17 11 20](https://user-images.githubusercontent.com/1945542/86009322-abc02a80-ba4c-11ea-8159-85f243aed3b7.png)
Search - dark | ![Simulator Screen Shot - iPhone 11 - 2020-06-29 at 17 02 40](https://user-images.githubusercontent.com/1945542/86009568-f5107a00-ba4c-11ea-8ace-0d453132fedb.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-29 at 17 11 49](https://user-images.githubusercontent.com/1945542/86009581-f8a40100-ba4c-11ea-84b1-a6b462a1356e.png)
Product variations - light | ![Simulator Screen Shot - iPhone 11 - 2020-06-29 at 17 03 11](https://user-images.githubusercontent.com/1945542/86009708-1c674700-ba4d-11ea-9a21-effe335ee2cd.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-29 at 17 11 34](https://user-images.githubusercontent.com/1945542/86009716-1fface00-ba4d-11ea-910c-8712f2046051.png)





Update release notes:
